ocdskit/oc4ids_transforms.py

```diff
@@ -0,0 +1,1010 @@
+
+        self.parties = [party["party"] for party in unique_parties]
+
+
+def copy_party_to_party_list(state, party):
+    output_parties = state.output.get("parties", [])
+    if not output_parties:
+        state.output["parties"] = output_parties
+
+    output_party = None
+    for party_to_match in output_parties:
+        if party.get("id") == party_to_match.get("id"):
+            output_party = party_to_match
+
+    if not output_party:
+        output_party = copy.deepcopy(party)
+        output_parties.append(output_party)
+    return output_party
+
+
+def copy_party_by_role(state, role, new_roles=None):
+    for party in state.parties:
+        if role in check_type(party.get("roles"), list):
+            output_party = copy_party_to_party_list(state, party)
+            if new_roles:
+                output_roles = output_party.get("roles", [])
+                output_roles.extend(new_roles)
+                output_party["roles"] = list(set(output_roles))
+
+
+def copy_document(state, document):
+    """
+    Copies a document. If it finds clasing ids changed ids to autoincrement numbers
+    """
+
+    output_documents = state.output.get("documents")
+    if not output_documents:
+        output_documents = []
+        state.output["documents"] = output_documents
+
+    duplicate_doc_id = False
+
+    for output_document in output_documents:
+        if output_document.get("id") == document.get("id"):
+            duplicate_doc_id = True
+            break
+
+    output_documents.append(document)
+
+    if duplicate_doc_id:
+        for num, doc in enumerate(output_documents):
+            doc["id"] = str(num + 1)
+
+
+def copy_document_by_type(state, document_type):
+
+    """
+    Copies documents of specific documentType from planning.documents to documents
+    """
+
+    for compiled_release in state.compiled_releases:
+        documents = resolve_pointer(compiled_release, "/planning/documents", [])
+        for document in check_type(documents, list):
+            document = check_type(document, dict)
+            if document_type == document.get("documentType"):
+                copy_document(state, document)
+
+
+def concat_ocid_and_string(state, path_to_string):
+
+    """
+    Places the ocid of a release in front of a string (eg. description or title)
+    so that it can be joined unambiguously with others, separated by new lines
+    """
+
+    strings = ""
+    for compiled_release in state.compiled_releases:
+
+        ocid = check_type(resolve_pointer(compiled_release, "/ocid", None), str)
+        a_string = check_type(resolve_pointer(compiled_release, path_to_string, None), str)
+
+        if a_string:
+            concat = "<{}> {}\n".format(ocid, a_string)
+            strings = strings + concat
+
+    return strings
+
+
+def public_authority_role(state):
+    """
+    CoST IDS element: Project owner
+    """
+    return copy_party_by_role(state, "publicAuthority")
```

Comment:
My understanding of the API is that these methods don't need to return anything. In that case, please remove return statements, unless they're used to quickly exit a transform.